vllm/core/block_manager.py

```diff
@@ -342,10 +354,13 @@ def swap_in(self, seq_group: SequenceGroup) -> Dict[int, int]:
                 if cpu_block in mapping:
                     gpu_block = mapping[cpu_block]
                     gpu_block.ref_count += 1
-                else:
+                elif self.enable_caching:
```

Comment:
Nit: I don't think this check is necessary. There's no hash being computed and the same check is in allocate.